pkg/internal/common/logging/modlog/deflog.go

```diff
@@ -0,0 +1,145 @@
+)
+
+const (
+	logLevelFormatter   = "UTC %s-> %s "
+	logPrefixFormatter  = " [%s] "
+	callerInfoFormatter = "- %s "
+)
+
+//defLog is a standard default logger implementation
+type defLog struct {
+	logger *log.Logger
+	module string
+}
+
+//Fatalf is CRITICAL log formatted followed by a call to os.Exit(1).
+func (l *defLog) Fatalf(format string, args ...interface{}) {
+	opts := metadata.GetLoggerOpts(l.module, logapi.CRITICAL)
+	l.logf(opts, logapi.CRITICAL, format, args...)
+	os.Exit(1)
+}
+
+//Panicf is CRITICAL log formatted followed by a call to panic()
+func (l *defLog) Panicf(format string, args ...interface{}) {
+	opts := metadata.GetLoggerOpts(l.module, logapi.CRITICAL)
+	l.logf(opts, logapi.CRITICAL, format, args...)
+	panic(fmt.Sprintf(format, args...))
+}
+
+//Debugf calls go 'log.Output' and can be used for logging verbose messages.
+// Arguments are handled in the manner of fmt.Printf.
+func (l *defLog) Debugf(format string, args ...interface{}) {
+	opts := metadata.GetLoggerOpts(l.module, logapi.DEBUG)
+	if !opts.LevelEnabled {
+		return
+	}
+	l.logf(opts, logapi.DEBUG, format, args...)
+}
+
+//Infof calls go 'log.Output' and can be used for logging general information messages.
+//INFO is default logging level
+// Arguments are handled in the manner of fmt.Printf.
+func (l *defLog) Infof(format string, args ...interface{}) {
+	opts := metadata.GetLoggerOpts(l.module, logapi.INFO)
+	if !opts.LevelEnabled {
+		return
+	}
+	l.logf(opts, logapi.INFO, format, args...)
+}
+
+// Warnf calls go log.Output and can be used for logging possible errors.
+// Arguments are handled in the manner of fmt.Printf.
+func (l *defLog) Warnf(format string, args ...interface{}) {
+	opts := metadata.GetLoggerOpts(l.module, logapi.WARNING)
+	if !opts.LevelEnabled {
+		return
+	}
+	l.logf(opts, logapi.WARNING, format, args...)
+}
+
+// Errorf calls go 'log.Output' and can be used for logging errors.
+// Arguments are handled in the manner of fmt.Printf.
+func (l *defLog) Errorf(format string, args ...interface{}) {
+	opts := metadata.GetLoggerOpts(l.module, logapi.ERROR)
+	if !opts.LevelEnabled {
+		return
+	}
+	l.logf(opts, logapi.ERROR, format, args...)
+}
+
+//ChangeOutput for changing output destination for the logger.
+func (l *defLog) ChangeOutput(output io.Writer) {
+	l.logger.SetOutput(output)
+}
+
+func (l *defLog) logf(opts *metadata.LoggerOpts, level logapi.Level, format string, args ...interface{}) {
+	//Format prefix to show function name and log level and to indicate that timezone used is UTC
+	customPrefix := fmt.Sprintf(logLevelFormatter, l.getCallerInfo(opts), metadata.ParseString(level))
+	err := l.logger.Output(2, customPrefix+fmt.Sprintf(format, args...))
+	if err != nil {
+		fmt.Printf("error from logger.Output %v\n", err)
+	}
+}
+
+func (l *defLog) getCallerInfo(opts *metadata.LoggerOpts) string {
+
+	if !opts.CallerInfoEnabled {
+		return ""
+	}
+
+	const MAXCALLERS = 6  // search MAXCALLERS frames for the real caller
```

Comment:
It would be helpful to further comment on why these particular values for the contents (i.e., MAXCALLERS = 6 and SKIPCALLERS = 4).

Reply:
Will fix it

Reply:
fixed it